db/version_set_test.cc

```diff
@@ -758,216 +763,397 @@ TEST_F(VersionSetTest, SameColumnFamilyGroupCommit) {
   EXPECT_EQ(kGroupSize - 1, count);
 }
 
-TEST_F(VersionSetTest, HandleValidAtomicGroup) {
-  std::vector<ColumnFamilyDescriptor> column_families;
-  SequenceNumber last_seqno;
-  std::unique_ptr<log::Writer> log_writer;
-  PrepareManifest(&column_families, &last_seqno, &log_writer);
-
-  // Append multiple version edits that form an atomic group
-  const int kAtomicGroupSize = 3;
-  std::vector<VersionEdit> edits(kAtomicGroupSize);
-  int remaining = kAtomicGroupSize;
-  for (size_t i = 0; i != edits.size(); ++i) {
-    edits[i].SetLogNumber(0);
-    edits[i].SetNextFile(2);
-    edits[i].MarkAtomicGroup(--remaining);
-    edits[i].SetLastSequence(last_seqno++);
-  }
-  Status s;
-  for (const auto& edit : edits) {
-    std::string record;
-    edit.EncodeTo(&record);
-    s = log_writer->AddRecord(record);
-    ASSERT_OK(s);
+class VersionSetAtomicGroupTest : public VersionSetTestBase,
+                                  public testing::Test {
+ public:
+  const static int kTestVersionSetRecover;
+  const static int kTestReactiveVersionSetRecover;
+  const static int kTestReactiveVersionSetReadAndApply;
+  VersionSetAtomicGroupTest() : VersionSetTestBase() {}
+
+  void SetUp() override {
+    PrepareManifest(&column_families_, &last_seqno_, &log_writer_);
+    SetupTestSyncPoints();
```

Comment:
It's unlikely that all test cases can use `SetupTestSyncPoints` to set up test sync points and the dependency among the sync points. Different tests need different sync points and the dependencies among the test sync points also differ.

Reply:
Yes. I was putting sync points for each test but it generates duplicate codes. Most of the tests here share the same set of sync points.

Reply:
I am OK with this. We can always change this in the future.